report.js

```diff
@@ -235,6 +263,13 @@ async function reportPostInstall () {
     return Promise.reject(new Error('No parent found, nothing to report'))
   }
 
+
+  if (allowTopLevel(dependencyInfo.rootPackage)) {
+    const gitSha = await getGitSha()
+    logIfVerbose(`Injecting sha to parent: ${gitSha}`)
+    dependencyInfo.parent.gitSha = gitSha
```

Comment:
`allowTopLevel` on its own does not guarantee that `parent == root`. I think this logic would work if `skipTraversal && allowTopLevel` but we probably need to be a little more general.

Reply:
i think then i should actually update this to make use of parentIsRoot and fix up parentIsRoot to be safe to use with optionals, realizing that the sha can only be retrieved from the actual root package.

Reply:
i updated the condition now that it should check parent is root and allowTopLevel is enabled. i updated the definition of parentIsRoot above to be safe: https://github.com/scarf-sh/scarf-js/pull/61/files#diff-7f51467a2b2b5a7527294c07a9a35c8f3195ad9e4ced1ab2c5d1450a75840713R77